### PR TITLE
Remove wrong internal assert in handle_view_on_rebase

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85849
* #79056
* #88216
* __->__ #88243

Fixes: https://github.com/pytorch/pytorch/issues/88205

The `CreationMeta::NO_GRAD_MODE` path in handle_view_on_rebase wrongly assumes that the tensor would be a leaf, because tensors created in no_grad are always leaf tensors. However, due to creation_meta propagation, a view of a view created in no_grad also has `CreationMeta::NO_GRAD_MODE`, but DOES have grad_fn.

### CODE REVIEW

File: test/test_autograd.py [hunk 1]
```diff
@@ -5881,6 +5881,15 @@ def run_tests(fn):
         run_tests(lambda v: v.swapdims_(0, 0))
         run_tests(lambda v: v.swapaxes_(0, 0))
 
+    def test_autograd_inplace_view_of_view(self):
+        x = torch.zeros(2, requires_grad=False)
+        with torch.no_grad():
+            y = x.view(2)
+            y.requires_grad_(True)
```
Comment (albanD) [inline test/test_autograd.py:5888]: Does this line need to be inside the no_grad block?

File: torch/csrc/autograd/variable.cpp [hunk 2]
```diff
@@ -789,7 +789,6 @@ void handle_view_on_rebase(
           " allow the output views to be modified inplace. You should replace the inplace operation by an"
           " out-of-place one.");
     } else if (creation_meta == CreationMeta::NO_GRAD_MODE) {
-      TORCH_INTERNAL_ASSERT(!grad_fn);
```
Comment (albanD) [inline torch/csrc/autograd/variable.cpp:792]: Actually could we update the error message to be even more precise on this case: you did a differentiable view on a view created in a no_grad block and are now modifying inplace.

**Review by albanD — COMMENTED** [review]
Nice catch!

File: torch/csrc/autograd/variable.cpp (hunk 2 above)
Comment (soulitzer) [reply torch/csrc/autograd/variable.cpp:792]: Good idea, updated.

**Review by albanD — APPROVED** [review]

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**